sky/utils/resources_utils.py

```diff
@@ -196,18 +197,36 @@ def format_resource(resource: 'resources_lib.Resources',
         elements_simple.append(f'gpus={acc}:{count}')
         elements_full.append(f'gpus={acc}:{count}')
 
-    if (resource.accelerators is None or is_k8s):
-        if vcpu is not None:
-            elements_simple.append(f'cpus={int(vcpu)}')
-            elements_full.append(f'cpus={int(vcpu)}')
-        if mem is not None:
-            elements_simple.append(f'mem={int(mem)}')
-            elements_full.append(f'mem={int(mem)}')
+    cpu_to_add = None
+    if resource.cpus is not None and resource.cpus.strip():
```

Comment:
After further checking, I think we should not use the `resource.cpus`, which is the resource that users specified when launching a cluster or job, and may not be the final resources.
For example, launch a cluster with `--cpus 0.1+`, the `resource.cpus` is `0.1+` and the `resource.memory` is `None`.
So we'd better just use the vcpu and mem as we have done previously.